src/tools/store.cpp

```diff
@@ -87,8 +91,22 @@ bool generateDataStoreOptions(const int argc,
     return true;
 }
 
+[[ noreturn ]] void CleanupSharedBarriers(int signum)
+{ // Here the lock state of named mutexes is unknown, make a hard cleanup
+    osrm::storage::SharedBarriers::resetCurrentRegions();
+    osrm::storage::SharedBarriers::resetRegions1();
```

Comment:
This is potentially dangerous:

These locks govern the shared mutexes for the data region. They should only be reset if osrm-datastore actually had a lock on them.

If these locks are just removed while a libosrm instance is running this can lead to crashes on the next data update or exceptions because locks will try to unlock mutexes that are in an unlocked state.